cocos/3d/skeletal-animation/skeletal-animation-state.ts

```diff
@@ -168,8 +162,24 @@ export class SkeletalAnimationState extends AnimationState {
     private _sampleCurvesBaked (time: number) {
         const ratio = time / this.duration;
         const info = this._animInfo!;
+
+        // Ensure I'm the one on which the anim info is sampling.
+        if (!this._animInfoMgr.isCurrentlySampling(info, this.clip)) {
+            // If not, switch to me.
+            this._animInfoMgr.switchClip(this._animInfo!, this.clip);
+
+            const users = this._parent!.getUsers();
+            users.forEach((user) => {
+                user.uploadAnimation(this.clip);
+            });
+        }
+
         const curFrame = (ratio * this._frames + 0.5) | 0;
-        if (curFrame === info.data[0]) { return; }
```

Comment:
@YunHsiao Please review the fix here. This cause the first frame always be skipped.